File: georocket-server-api/src/main/java/io/georocket/index/xml/MetaIndexerFactory.java
package io.georocket.index.xml;

import io.georocket.index.IndexerFactory;

/**
 * Factory for {@link MetaIndexer} objects
 * @since 1.0.0
 * @author Michel Kraemer
 */
public interface MetaIndexerFactory extends IndexerFactory {
  /**
   * @return a new instance of {@link MetaIndexer}
   */
  @Override
  MetaIndexer createIndexer();
}

File: georocket-server/src/main/kotlin/io/georocket/index/MetaIndexerFactory.kt
package io.georocket.index

import io.georocket.query.QueryCompiler

/**
 * Factory for [MetaIndexer] objects
 * @author Michel Kraemer
 */
interface MetaIndexerFactory : QueryCompiler {
  /**
   * Returns a new instance of [MetaIndexer]
   */
  fun createIndexer(): MetaIndexer
}
